File: Kattis/solutions/eenyMeeny/eenyMeeny.cpp
#include <iostream>
#include <queue>
#include <vector>
#include <sstream>

using namespace std;

int main() {
  string line;
  getline(cin, line);
  stringstream ss(line);
  vector<string> rhymes;
  string word;
  while (ss >> word) {
    rhymes.push_back(word);
  }

  int numKids;
  cin >> numKids;
  cin.ignore();

  queue<string> kidsQueue;
  for (int i = 0; i < numKids; i++) {
    string kid;
    cin >> kid;
    kidsQueue.push(kid);
  }

  vector<string> team1;
  vector<string> team2;
  bool team = true;

  while (!kidsQueue.empty()) {
    for (int i = 0; i < rhymes.size() - 1; i++) {
      string kid = kidsQueue.front();
      kidsQueue.pop();
      kidsQueue.push(kid);
    }
    if (team) {
      team1.push_back(kidsQueue.front());
    } else {
      team2.push_back(kidsQueue.front());
    }
    kidsQueue.pop();
    team = !team;
  }

  cout << team1.size() << "\n";
  for (auto& kid : team1) {
    cout << kid << "\n";
  }

  cout << team2.size() << "\n";
  for (auto& kid : team2) {
    cout << kid << "\n";
  }

  return 0;
}

File: Kattis/solutions/eenyMeeny/eenyMeeny.java
import java.util.*;

class eenyMeeny {
  public static void main(String[] args) {
    Scanner s = new Scanner(System.in);
    String line = s.nextLine();
    String[] rhymes = line.split(" ");
    int numKids = s.nextInt();
    s.nextLine();
    
    Queue<String> queue = new LinkedList<>();
    for (int i = 0; i < numKids; i++) {
      String kid = s.nextLine();
      queue.add(kid);
    }
    
    ArrayList<String> team1 = new ArrayList<>();
    ArrayList<String> team2 = new ArrayList<>();
    boolean team = true;
    while (!queue.isEmpty()) {
      for (int i = 0; i < rhymes.length - 1; i++) {
        String kid = queue.remove();
        queue.add(kid);
      }
      if (team) {
        team1.add(queue.remove());
      } else {
        team2.add(queue.remove());
      }
      team = !team;
    }
    
    System.out.println(team1.size());
    for (String kid : team1) {
      System.out.println(kid);
    }
    
    System.out.println(team2.size());
    for (String kid : team2) {
      System.out.println(kid);
    }
        
  }
}
